src/librustdoc/html/render/print_item.rs

```diff
@@ -1585,61 +1599,92 @@ fn item_constant(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, c: &cle
 }
 
 fn item_struct(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, s: &clean::Struct) {
-    wrap_item(w, |w| {
-        render_attributes_in_code(w, it, cx.tcx());
-        render_struct(w, it, Some(&s.generics), s.ctor_kind, &s.fields, "", true, cx);
-    });
+    item_template!(
```

Comment:
Please move this struct definition and implementation outside of the function so it's easier to read (less lines of code in the function and less indent for the struct implementation). Just add a comment on `ItemStruct` and `Field` that they are only used in `item_struct` (and inside the template for `Field`).